### DISCUSSION TITLE
Some questions regarding custom sources

### DISCUSSION THREAD

**Discussion by jan-dh** [craftcms/cms#13434] [Q&A] [answered]
1. Is it possible to "preview" a custom source? I link it to a specific user group, but that means I always have to log in as a user of that group to preview. 
1. It's somewhat unclear UX-wise, when you choose multiple entry criteria if the entries shown are the ones that match BOTH criteria or just ONE
1. Are there any plans of adding the "Entry ID" as a possible filter option? Would love to pass an array of a couple of ideas to the filter

I'm stuck in the scenario where I have 1 main structure for a website and a lot of team specific structures. I'm looking for ways to:
1. Give each team a user group that has acces the entries specific for their team in the main structure - since I can't filter on ID's this is not really possible (slugs can be changed)
2. Give them access to their own structure - this is currently already possible

**Comment by brandonkelly** [comment, marked as answer]
> Is it possible to "preview" a custom source? I link it to a specific user group, but that means I always have to log in as a user of that group to preview.

Admin users will see all sources.

> It's somewhat unclear UX-wise, when you choose multiple entry criteria if the entries shown are the ones that match BOTH criteria or just ONE

All condition rules will be applied, just like element query params (`craft.entries().section('news').authorGroup('editors')` will return entries that are in the `news` section AND whose authors are in the user group `editors`).

> Are there any plans of adding the "Entry ID" as a possible filter option? Would love to pass an array of a couple of ideas to the filter

That’s not possible currently because element IDs are likely to differ from environment to environment. We should be able to make it possible to set the value to an environment variable though, as we did with relation condition rules, but that will be a breaking change that would need to wait for Craft 5.

> I'm stuck in the scenario where I have 1 main structure for a website and a lot of team specific structures.

Do you mean that each team has their own entire Structure section? Or just a subset of entries within a single larger Structure section?

**Comment by jan-dh** [comment]
I mean each team has their own entire structure section. It's not my favorite route to go, but I need the ability to "contain" user profiles to limit them to the content relevant for them. Putting it all in 1 structure would be my preferred way to go, but there are no options to limit user profiles to entries, only to sections, so it's a matter of working with what we have. Was seriously contemplating in duplicating entry types just for the sake of being able to use them in Custom sources, which again, feels like a dirty work around. I guess my best guess for now is to to create a dropdown list in the main structure, only visible for admins and use that dropdown to select the right "team" bucket, then use that value in a Custom source.  There used to be a plugin that got me a little closer: https://plugins.craftcms.com/isolate?craft4 but this one is not maintained anymore.

**Reply by brandonkelly** [reply]
You could add a “Team” dropdown field to the main section (or Checkboxes, if some entries should be shared by multiple teams), and then create custom sources that pull in entries from that section, each limited to only show entries where the custom field includes the proper team value. The custom field could be limited to only allow admins to view (and edit) it, so team members can’t go in and tamper with it.

**Comment by jan-dh** [comment]
@brandonkelly would it be possible to add the structure view type of all the entries were in fact in a structure section? Kind of hard to work with multi-level entries this way.

**Reply by brandonkelly** [reply]
Maybe… do you mind posting a new “Ideas” discussion about that?
